### ISSUE TITLE
Problem when using Type::build('datetime')->useLocaleParser();

### ISSUE THREAD

**Issue by gmponos** [cakephp/app#291]
Basically I had the same issue with this guy on this question: http://stackoverflow.com/questions/32768919/cakephp3-newentity-nulls-date.

So to take it from the top I have a field `appointment` in the database which is datetime type. I use an input in a form which is type text for this field. When I submit the form after the patchEntity the field is returning null. The function that returns the null is https://github.com/cakephp/cakephp/blob/master/src/Database/Type/DateTimeType.php#L232  but I can't really figure out why.

If I remove the following lines https://github.com/cakephp/app/blob/master/config/bootstrap.php#L202 the field is not null but it is not an object. It is a string and can be saved successfully in the database.

If I use what is suggested here: http://book.cakephp.org/3.0/en/core-libraries/internationalization-and-localization.html#parsing-localized-datetime-data and set a local format `setLocaleFormat('dd-M-y')` the field after the patch it is returning as an object and can be saved successfully.

So right now I am a little bit confused... Is this a bug? Is this something that should be mentioned on the installation process or maybe should set a default locale inside bootstrap with `setLocaleFormat()` function in order to be more obvious that someone should change it to their needs and also document it with a proper DocBlock? Am I doing something wrong?

**Comment by ADmad** [comment]
It's not a bug. In order for a datetime string to be properly marshaled into an object, the format in which you pass the string should match the format set for the database type set for the table column.

First thing, is the `appointment` field of type `date` or `datetime` in your database? Based on info you have provided it's a `datetime` field while you are passing a date string for it, just the format does not match and it gets marshaled to `null`.

**Comment by gmponos** [comment]
OK... I did the following:
- changed the type of the field to date,
- changed the validation rule to date,
- tried to save a value 2015-10-05.

Everything worked.

- changed the type of the field to datetime,
- changed the validation rule to datetime,
- tried to save a value 2015-10-06 10:10.
 returns null

**Comment by gmponos** [comment]
Yea you are right double checked it and works.

**Comment by gmponos** [comment]
Inside boostrap.php I have the default settings for the following lines:

    ini_set('intl.default_locale', 'en_US');
    Type::build('datetime')->useLocaleParser();

1. If I write `2015-10-10 10:10 PM` the validation passes but the date is saved as `11/10/72, 10:10 PM` so the defaults comes with `en_US` which requires `mdy` but the validation default requires `ymd`.
2. If I change the validation of datetime to `mdy` everything is saved correctly. When I edit my record I get a comma between date and time inside the text field and I can't submit the form because it doesn't pass validation. So each time I have to remove this comma.
3. If I change my locale to `el_GR` the `PM` is the greek letters `μ.μ.`. So If I write inside the input `2015-10-10 10:10 μ.μ.` it doesn't pass validation. If I skip this and don't write the greek `μ.μ.` and I write `2015-10-10 20:10` the validation passes but the field after patchEntity function is returned as null.

**Comment by markstory** [comment]
Seems like the validator for mdy needs to accept a comma between the dat and time parts.

**Comment by markstory** [comment]
Pull request to fix the validator method up now.

**Comment by gmponos** [comment]
How about the 1 and 3 from my previous comment? I believe that cakephp/app should come with a default `Type::build('datetime')->useLocaleParser()->setLocaleFormat('Y-m-d H:i');` in order not to have things like that.

**Comment by lorenzo** [comment]
That would not be a locale parser, would it? A fixed format is not dependent of the current locale, which is the intention of that line

**Comment by gmponos** [comment]
First, I am just saying have a default one that doesn't confuse things up. It took me a while to figure out that because I had set the locale to `el_GR` and I don't commit `μ.μ.` it returns null. 

Second, of all and continuing from the first as you said what is the use of locale parser for me when I set my locale to `el_GR` and because I can't submit the `μμ` I am retrieving a null object?

Read the **3** from my previous comment.
